src/NuGet.Clients/NuGet.CommandLine/Commands/DeprecatedCommandAttribute.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace NuGet.CommandLine
+{
+    /// <summary>
+    /// Attribute for marking nuget.exe commands as deprecated.
+    ///
+    /// By using this attribute in ICommand-based classes, it will show a warning message each
+    /// command invocation and in each help command invocation.
+    /// 
+    /// <see cref="ICommand"/>
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = false)]
+    public sealed class DeprecatedCommandAttribute : Attribute
+    {
+        /// <summary>
+        /// The concrete type that is an alternative to the command that has this DeprecatedCommand attribute.
+        /// If no alternative is provided in the attribute, this property will be null
+        /// </summary>
+        public Type AlternativeCommand { get; set; }
+
+        public DeprecatedCommandAttribute()
+        {
+            AlternativeCommand = null;
+        }
+
+        public DeprecatedCommandAttribute(Type alternativeCommand)
+        {
+            if (!typeof(ICommand).IsAssignableFrom(alternativeCommand))
```

Comment:
This block would probably be better in the `AlternativeCommand` getter, otherwise since you also have a default constructor, people could just use `[DeprecatedCommand(AlternativeCommand = typeof(string))]`.

Reply:
i mean setter, the block should be in the property setter